firebase/functions/index.js

```diff
@@ -5,84 +5,70 @@ const functions = require('firebase-functions')
-    const totalTimeSpentMappingRef          = admin.database().ref('/v2/users/'+context.params.userId+'/timeSpentMapping')
+    const groupRef = admin.database().ref('/v2/groups/' + context.params.projectId + '/' + context.params.groupId)
+    const userRef = admin.database().ref('/v2/users/' + context.params.userId)
+    const userContributionRef = userRef.child('contributions/' + context.params.projectId)
 
-    const startTimeRef          = admin.database().ref('/v2/results/'+context.params.projectId+'/'+context.params.groupId+'/'+context.params.userId+'/startTime')
-    const endTimeRef            = admin.database().ref('/v2/results/'+context.params.projectId+'/'+context.params.groupId+'/'+context.params.userId+'/endTime')
-    const timeSpentMappingRef   = admin.database().ref('/v2/results/'+context.params.projectId+'/'+context.params.groupId+'/'+context.params.userId+'/timeSpentMappingRef')
-
-    // Counter for users
-    const groupContributionCount = groupContributionCountRef.transaction((currentCount) => {
+    // Counter for User
+    const totalGroupContributionCount = userRef.child('groupContributionCount').transaction((currentCount) => {
         return currentCount + 1
     });
-    promises.push(groupContributionCount)
+    promises.push(totalGroupContributionCount)
 
-    const taskContributionCount = numberOfTasksRef.once('value')
+    const totalTaskContributionCount = groupRef.child('numberOfTasks').once('value')
         .then((dataSnapshot) => {
             const numberOfTasks = dataSnapshot.val()
             return numberOfTasks
         })
         .then((numberOfTasks) => {
-            return taskContributionCountRef.transaction((currentCount) => {
+            return userRef.child('taskContributionCount').transaction((currentCount) => {
                 return currentCount + numberOfTasks
             })
         })
-    promises.push(taskContributionCount)
+    promises.push(totalTaskContributionCount)
 
-    // Counters for tasks and groups per user and per project
-    const projectGroupContributionCount = projectGroupContributionCountRef.transaction((currentCount) => {
+    // Counter for User per Project
+    const groupContributionCount = userContributionRef.child('groupContributionCount').transaction((currentCount) => {
         return currentCount + 1
     });
-    promises.push(projectGroupContributionCount)
+    promises.push(groupContributionCount)
 
-    const projectTaskContributionCount = numberOfTasksRef.once('value')
+    const taskContributionCount = groupRef.child('numberOfTasks').once('value')
         .then((dataSnapshot) => {
             const numberOfTasks = dataSnapshot.val()
             return numberOfTasks
         })
         .then((numberOfTasks) => {
-            return projectTaskContributionCountRef.transaction((currentCount) => {
+            return userContributionRef.child('taskContributionCount').transaction((currentCount) => {
```

Comment:
this looks duplicated. What's the difference to the code in line 41 above?

Reply:
probably here you wanted to add the number of tasks done for a specific project?

Reply:
ah or it's my mistake and I didn't check the difference between `UserRef` and `UserContributionRef`...